product_multi_barcode/views/product_template_view.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="product_template_form_view_multi_barcode" model="ir.ui.view">
+        <field name="name">product.template.form.multi.barcode</field>
+        <field name="model">product.template</field>
+        <field name="priority">16</field>
+        <field name="inherit_id" ref="product.product_template_only_form_view" />
+        <field name="arch" type="xml">
+            <field name="barcode" position="attributes">
+                <attribute name="readonly">True</attribute>
+            </field>
+            <field name="barcode" position="after">
+                <field
+                    name="barcode_ids"
+                    attrs="{'invisible': [('product_variant_count', '>', 1)]}"
```

Comment:
```suggestion
```
Why product_variant_count > 1?
Wouldn't be want it in single variant templates too?

Reply:
mmm, we only show barcode_ids if only single variant, I've used same criteria of odoo  with [barcode field](https://github.com/odoo/odoo/blob/13.0/addons/product/views/product_template_views.xml#L39)

(it's invisible when more than one variant)